sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -969,9 +970,10 @@ object DateTimeUtils {
   def currentTimestamp(): Long = instantToMicros(Instant.now())
 
   /**
-   * Obtains the current local date-time as microseconds since the epoch in the default time-zone.
+   * Obtains the current local date-time as microseconds since the epoch at the session time-zone.
    */
-  def currentTimestampNTZ(): Long = localDateTimeToMicros(LocalDateTime.now())
+  def currentTimestampNTZ(): Long = localDateTimeToMicros(
+    LocalDateTime.now(getZoneId(SQLConf.get.sessionLocalTimeZone)))
```

Comment:
Please, extend `LocalTimestamp` by `TimeZoneAwareExpression`, and take the time zone id from there.